### PR TITLE
Upgrade script bugs

### PR DESCRIPTION
**Pull request by amitaekbote**
## High Level Description

Fixes bug in the upgrade script

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [x] Test Results: [link to CI job test results for component]
  - [x] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@branden @darkonie ptal

File: dcos_installer/upgrade.py [hunk 1]
```diff
@@ -74,13 +74,17 @@
 # Check if the node has node/cluster checks and run them
 if [ -f /opt/mesosphere/etc/dcos-diagnostics-runner-config.json ]; then
    # command exists
-   if !(output=$(dcos-diagnostics check node-poststart)); then
+   output=$(dcos-diagnostics check node-poststart)
+   ret=$?
+   if [ $ret -ne 0 ]; then
```
Comment (darkonie) [inline dcos_installer/upgrade.py]: just curious, is there a benefit of assigning `$?` to `ret`?
instead of just comparing:
`if [ $? -ne 0 ]; then`

Comment (greggomann) [inline dcos_installer/upgrade.py]: Ditto - can we remove the assignment?

Looking below in this file, we follow the same pattern of assignment as used here. Local consistency is nice - maybe either go with the existing style (the other occurrence uses `$RETCODE`), or add another commit to this PR which updates the other site as well?

File: dcos_installer/upgrade.py [hunk 2]
```diff
@@ -74,13 +74,17 @@
 # Check if the node has node/cluster checks and run them
 if [ -f /opt/mesosphere/etc/dcos-diagnostics-runner-config.json ]; then
```
Comment (greggomann) [inline dcos_installer/upgrade.py]: While you're touching these two conditional blocks, would you mind indenting them to 4 spaces? Looks like all other indented lines in this file use 4.

**Comment by amitaekbote** [comment]
Thanks @greggomann for verifying the fix, ill address the comments

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot  bump-ee

**Comment by amitaekbote** [comment]
@greggomann @darkonie addressed the comments, CI failure are the flaky ones

File: dcos_installer/upgrade.py [hunk 3]
```diff
@@ -73,19 +73,20 @@
...
-
-   if !(clusteroutput=$(dcos-diagnostics check cluster)); then
-      echo "Cannot proceed with upgrade, cluster checks failed"
-      echo $clusteroutput
-      exit 1
-   fi
+    # command exists
+    output=$(dcos-diagnostics check node-poststart)
+    if [ $? -ne 0 ]; then
+       echo "Cannot proceed with upgrade, node checks failed"
```
Comment (branden) [inline dcos_installer/upgrade.py]: This section is indented at 3 spaces, should be 4.

**Comment by amitaekbote** [comment]
@branden ptal

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label ship it

**Comment by jgehrcke** [comment]
@amitaekbote I'd love to close a JIRA ticket now that this is merged. Do we have one? If not: why don't we have one?

**Comment by amitaekbote** [comment]
It was a bug fix that I noticed, i created the jira a little late :) 

Here it is and its closed 
https://jira.mesosphere.com/browse/DCOS-16605

**Pull request merged**